### PR TITLE
Admin Update With Inline SVG's

### LINKED ISSUE #7584: Admin layout unusable in small windows (it isn't responsive)
My guess is that this isn't exactly a top priority task, but here you go...

Also, is this something that might be good if I opened a PR for if I have some time to?
- Left sidebar menu doesn't show (the main menu for admins))
- The hamburger menu moves the content on the screen left and right (probably because the left column is hidden from view, but still being manipulated)
- The title of the page is cut off
- Hamburger menu is flush to the right of the screen, right padding should probably match left padding of the logo

screen shot 2016-09-16 at 9 06 28 am

screen shot 2016-09-16 at 9 35 27 am

### PR DESCRIPTION
**Pull request by MatthewKennedy**
Use inline SVG's over CSS for better results.

In my opinion, this is the better way to go, as you are actually fixing the legacy problems left over from the different font based icon packs and individual naming conventions of each, plus with a little more work this could really shine, its easy to edit and add new icons.

Where as, the other PR, is less drastic on the surface, but really, it's just bodging the flattened SVG's in a CSS file. (Hard to edit the SVG's, no way to set color via css, the filter hack for black and white rasterises the SVG).

I'm happy to work on either if you wish, but didn't want to feel like I only offered one solution, when I wasn't really happy with the solution myself.

Fixes https://github.com/spree/spree/issues/7584 https://github.com/spree/spree/issues/8287

### CODE REVIEW

File: backend/spec/features/admin/orders/listing_spec.rb [hunk 1]
```diff
@@ -191,39 +191,6 @@
       end
     end
 
-    # regression tests for https://github.com/spree/spree/issues/6888
-    context 'per page dropdown', js: true do
```
Comment (damianlegawiec) [inline backend/spec/features/admin/orders/listing_spec.rb:195]: @MatthewKennedy can we leave the per page dropdown?

File: backend/app/helpers/spree/admin/navigation_helper.rb [hunk 2]
```diff
@@ -227,6 +230,7 @@ def configurations_sidebar_menu_item(link_text, url, options = {})
       end
 
       def main_part_classes
+        # DEPRECIATE IN SPREE 5.0
```
Comment (damianlegawiec) [inline backend/app/helpers/spree/admin/navigation_helper.rb:233]:
```suggestion
        ActiveSupport::Deprecation.warn(<<-DEPRECATION, caller)
          Admin::NavigationsHelper#main_part_classes is deprecated and will be removed in Spree 5.0.
        DEPRECATION
```

File: backend/app/helpers/spree/admin/navigation_helper.rb [hunk 3]
```diff
@@ -235,6 +239,7 @@ def main_part_classes
       end
 
       def main_sidebar_classes
+        # DEPRECIATE IN SPREE 5.0
```
Comment (damianlegawiec) [inline backend/app/helpers/spree/admin/navigation_helper.rb:242]:
```suggestion
        ActiveSupport::Deprecation.warn(<<-DEPRECATION, caller)
          Admin::NavigationsHelper#main_sidebar_classes is deprecated and will be removed in Spree 5.0.
        DEPRECATION
```

File: backend/app/helpers/spree/admin/navigation_helper.rb [hunk 4]
```diff
@@ -243,6 +248,7 @@ def main_sidebar_classes
       end
 
       def wrapper_classes
+        # DEPRECIATE IN SPREE 5.0
```
Comment (damianlegawiec) [inline backend/app/helpers/spree/admin/navigation_helper.rb:251]:
```suggestion
        ActiveSupport::Deprecation.warn(<<-DEPRECATION, caller)
          Admin::NavigationsHelper#wrapper_classes is deprecated and will be removed in Spree 5.0.
        DEPRECATION
```

**Review by damianlegawiec — COMMENTED** [review]
@MatthewKennedy this looks amazing, thank you! 👍 Could you review my last comment?

**Comment by MatthewKennedy** [comment]
@damianlegawiec I will go through the comments and make the requested changes.

**Review by AleksandraKaminska — APPROVED** [review]

File: backend/app/assets/javascripts/spree/backend/promotions.js [hunk 5]
```diff
@@ -147,7 +147,7 @@ function initProductActions () {
       var quantity = $container.find('input[name="add_quantity"]').val()
       if (variant_id) {
         // Add to the table
-        var newRow = '<tr><td>' + productName + '</td><td>' + quantity + '</td><td><img src="/assets/admin/icons/cross.png"/></td></tr>'
+        var newRow = '<tr><td>' + productName + '</td><td>' + quantity + '</td><td><img src="/assets/admin/icons/cross.png"/>KKK</td></tr>'
```
Comment (kaliszKrzysztof) [inline backend/app/assets/javascripts/spree/backend/promotions.js:150]: Is there a specific reason of putting `KKK` here? Also 
```suggestion
        var newRow = '<tr><td>' + product_name + '</td><td>' + quantity + '</td><td><img src="/assets/admin/icons/cross.png"/>KKK</td></tr>'
```

Is there a specific reason of putting `KKK` here?

**Review by kaliszKrzysztof — APPROVED** [review]

File: backend/app/assets/javascripts/spree/backend/promotions.js (hunk 5 above)
Comment (MatthewKennedy) [reply backend/app/assets/javascripts/spree/backend/promotions.js]: @kaliszKrzysztof I was looking for that missing png in the DOM, I couldn't find it so just swapped it out for a span bases icon.

**Comment by MatthewKennedy** [comment]
This change is going to cause some extensions to need updates, and some new icons adding along the way.

**Comment by damianlegawiec** [comment]
> This change is going to cause some extensions to need updates, and some new icons adding along the way.

Can we make sure that the old classes will also work? Just aliasing old CSS classes to new ones?

**Comment by MatthewKennedy** [comment]
> Can we make sure that the old classes will also work? Just aliasing old CSS classes to new ones?

@damianlegawiec The helper methods in this PR does not generate the span tags to accommodate css glyph icons, they now insert inline svg's directly into the buttons and links.

I'm going to do some experiments with a not found svg's, see if I can sett a fall back to generate span tags rather than an empty svg with a message showing what icon was not found, do you think this is a good idea?

**Comment by MatthewKennedy** [comment]
> > This change is going to cause some extensions to need updates, and some new icons adding along the way.
> 
> 
> 
> Can we make sure that the old classes will also work? Just aliasing old CSS classes to new ones?
> 
> 

Think I’ve got this covered! Wait one moment.

**Comment by MatthewKennedy** [comment]
@damianlegawiec I have had to override the placeholder method inside the inline_svg gem helpers.rb, 

The placeholder is not an empty SVG with warning test inside, but rather the span with the correct classes for the glyphicons to appear.

Do you know a better way to override just the one method with a decorator? Rather than replacing the whole file?

Also will this override cause an issue on the frontend? If they are both using the same gem?

**Comment by damianlegawiec** [comment]
> @damianlegawiec I have had to override the placeholder method inside the inline_svg gem helpers.rb,
> 
> The placeholder is not an empty SVG with warning test inside, but rather the span with the correct classes for the glyphicons to appear.
> 
> Do you know a better way to override just the one method with a decorator? Rather than replacing the whole file?
> 
> Also will this override cause an issue on the frontend? If they are both using the same gem?

This will cause issues with the frontend, unfortunately... Let's skip this failover for now

**Comment by MatthewKennedy** [comment]
@damianlegawiec One other option would be to have the admin helpers look for .svg in the icon name, if ends with .svg, use inline SVG mode, if not fallback to a span with appropriate classes?

### Inline SVG
Screenshot 2020-06-22 at 13 15 23
Screenshot 2020-06-22 at 13 15 41

### Fallback Version
Screenshot 2020-06-22 at 13 16 58
Screenshot 2020-06-22 at 13 15 55

What do you think to that solution?

**Comment by MatthewKennedy** [comment]
Fallback to glyphicons Option 2

This works, Just need to go through the admin and add .svg to the icons you want to use as inline svg

In this image I have added in the static content extension, as you can see its fallen bak to use the glyphicons leaf image in the main menu.

Screenshot 2020-06-22 at 14 26 00

**Comment by damianlegawiec** [comment]
@MatthewKennedy excellent, that works for us! Thank you!

**Comment by MatthewKennedy** [comment]
@damianlegawiec No problem, I’ll just go through the admin view files and get them using the svg’s as much as possible.

**Review by damianlegawiec — APPROVED** [review]
Thanks again @MatthewKennedy -  this is a huge milestone in terms of Admin Panel UX!

**Comment by MatthewKennedy** [comment]
> Thanks again @MatthewKennedy - this is a huge milestone in terms of Admin Panel UX!

@damianlegawiec - no problem at all.

**Pull request merged**